### PR TITLE
adds redirect configurations and probes to azure_rm_appgateway

### PR DESCRIPTION
**Pull request by mjhoffman65**
##### SUMMARY
Adds redirect configurations and probes to the azure_rm_appgateway module.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_appgateway

##### ANSIBLE VERSION
```
ansible 2.8.0.dev0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible-2.8.0.dev0-py2.7.egg/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
Allows custom health probes to be set on the Application Gateway which is useful if the gateway is fronting a Kubernetes cluster with Nginx Ingress.  Also, allows redirect configurations to be set allowing one to redirect HTTP to HTTPS for example.

### CODE REVIEW

**Comment by yuwzho** [comment]
@zikalino

File: changelogs/fragments/azure_rm_appgateway-probe.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
```
Comment (zikalino) [inline changelogs/fragments/azure_rm_appgateway-probe.yaml:1]: do we actually need changelog fragment here?

Comment (zikalino) [reply changelogs/fragments/azure_rm_appgateway-probe.yaml:1]: yes, actually seems like it was not enforced too much before, nor documented, but we should add doc fragments now

**Comment by mjhoffman65** [comment]
bot_status

**Comment by mjhoffman65** [comment]
bot_status

**Review by zikalino — APPROVED** [review]
I would do it in exactly the same way :-) so merging

File: changelogs/fragments/azure_rm_appgateway-probe.yaml (hunk 1 above)
Comment (gundalow) [reply changelogs/fragments/azure_rm_appgateway-probe.yaml:1]: Hi, just for info. We are discussing the cases when a changelog is and updating CI to error if it isn't present.
Generally speaking if you are updating something apart from a `./test/` or a typo in documentation a changelog fragment is useful, and makes it easier to backport fixes as changelogs are required there.

**Pull request merged**